resen/Resen.py

```diff
@@ -662,13 +662,56 @@ def create_container(self,**input_kwargs):
         if image_id not in local_image_ids:
             print("Pulling image: %s" % image_name)
             print("   This may take some time...")
-            self.docker.images.pull(pull_image)
+            DockerHelper.stream_pull_image(pull_image)
+            image = self.docker.images.get(pull_image)
+            repo,digest = pull_image.split('@')
+            # When pulling from repodigest sha256 no tag is assigned. So:
+            image.tag(repo, tag=image_name)
             print("Done!")
 
         container_id = self.docker.containers.create(image_id,**create_kwargs)
 
         return container_id.id
 
+    @staticmethod
+    def stream_pull_image(pull_image):
+        import datetime
+        def truncate_secs(delta_time, fmt=":%.2d"):
+            delta_str = str(delta_time).split(':')
+            return ":".join(delta_str[:-1]) + fmt%(float(delta_str[-1]))
+        def update_bar(sum_total,accumulated,t0,current_time, scale=0.5):
+            percentage = accumulated/sum_total*100
+            nchars = int(percentage*scale)
+            bar = "\r["+nchars*"="+">"+(int(100*scale)-nchars)*" "+"]"
+            time_info = "Elapsed time: %s"%truncate_secs(current_time - t0)
+            print(bar+" %5.2f %%, %5.3f/%4.2fGB %s"%(percentage,
+                accumulated/1024**3,sum_total/1024**3,time_info),end="")
+
+        client = docker.APIClient(base_url='unix://var/run/docker.sock')
```

Comment:
Is `/var/run/docker.sock` platform independent? I think so ([see here](https://stackoverflow.com/questions/36765138/bind-to-docker-socket-on-windows)), but we should keep this assumption in mind.

Reply:
I'll put a comment in the code.